src/plugins/dashboard/public/dashboard_top_nav/internal_dashboard_top_nav.tsx

```diff
@@ -313,10 +327,83 @@ export function InternalDashboardTopNav({
       });
     }
     if (showWriteControls && managed) {
-      allBadges.push(getManagedContentBadge(dashboardManagedBadge.getTooltip()));
+      const managedBadge = (
+        <button
+          ref={managedBadgeRef}
+          onClick={() => {
+            setIsPopoverOpen(!isPopoverOpen);
+          }}
+          css={css`
+            :hover {
+              cursor: pointer;
+            }
+          `}
+        >
+          {createBadge(getManagedContentBadge(dashboardManagedBadge.getText(), true), 3)}
```

Comment:
Very smart exposing `createBadge`!! 🤯 I think by doing this, we can get an even cleaner solution **without** using `EuiWrappingPopover` - I was able to get this down from ~60 lines to ~30 lines. Here's a summary of what I did:
1. To `allBadges`, push a badge with (1) `badgeText` and (2) `renderCustomBadge`
2.  In the `renderCustomBadge`, build the "button" of the popover via the `createBadge` call. For the `TopNavMenuBadgeProps` argument, spread the result of `getManagedContentBadge` and add the things necessary to make the badge clickable - i.e. `onClick`
3. Return an `EuiPopover` component with the badge from (2) provided as the `button` prop

In the `renderCustomBadge` callback, make sure to use the `badgeText` argument that is provided as described in https://github.com/elastic/kibana/pull/189404#discussion_r1700328195.

Also, what do we think about this for the design?
 
| Current | Suggested |
|--------|--------|
| ![image](https://github.com/user-attachments/assets/b4e31a3c-543a-4f39-875f-cef38c7bdd51) | ![image](https://github.com/user-attachments/assets/d93468e9-ff59-435d-a14d-9d9196d90c01) |

Reply:
changed! thanks!